event/consumer.go

```diff
@@ -9,61 +12,99 @@ import (
 // MessageConsumer provides a generic interface for consuming []byte messages (from Kafka)
 type MessageConsumer interface {
 	Incoming() chan kafka.Message
-	Closer() chan bool
 }
 
 // Handler represents a handler for processing a batch of events.
 type Handler interface {
 	Handle(events []*ObservationExtracted) error
 }
 
+type Consumer struct {
+	closing chan bool
+	closed  chan bool
+}
+
+// NewConsumerGroup returns a new consumer group using default configuration.
+func NewConsumer() *Consumer {
+
+	c := Consumer{
```

Comment:
(Picky I know) but couldn't you just return &Consumer{...}

Reply:
done 👍